transport.go

```diff
@@ -95,27 +96,34 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	return resp, err
 }
 
-// Token checks the active token expiration and renews if necessary. Token returns
+// TokenWithContext is the same as Token with the ability to pass a Context.
+// Checks the active token expiration and renews if necessary. Returns
 // a valid access token. If renewal fails an error is returned.
-func (t *Transport) Token() (string, error) {
+func (t *Transport) TokenWithContext(ctx context.Context) (string, error) {
```

Comment:
When context support was added to some places in standard library, they used just a `Context` suffix (rather than `WithContext`). E.g:

- https://godoc.org/os/exec#CommandContext
- https://godoc.org/net#Dialer.DialContext
- https://godoc.org/database/sql#Conn.ExecContext
- https://godoc.org/database/sql#Conn.PingContext

So maybe consider naming this method `TokenContext` to follow the same pattern.

Also, question, is the only reason you're adding a new method to avoid making a breaking API change?